src/software/firmware/srcs/respirator.cpp

```diff
@@ -43,6 +47,10 @@ HardwareTimer* hardwareTimer3;
 Blower* blower_pointer;
 Blower blower;
 
+#if HARDWARE_VERSION == 2
+HardwareSerial Serial6(PIN_SERIAL_RX, PIN_SERIAL_TX);
```

Comment:
I think, there is an issue with `PIN_SERIAL_RX` and `PIN_SERIAL_TX ` usage.
They are defined in the variant of the STM32 core and used for the default instantiated `Serialx` mapped to the generic name `Serial`
https://github.com/stm32duino/Arduino_Core_STM32/blob/16f3644e7b9f8225dcb01f48bd857bdc3a978580/variants/NUCLEO_F411RE/variant.h#L105-L110

In the case of Nucleo F411RE, this instantiate Serial1 (USART1) using PA3/PA2.

For the HARDWARE V2, those definitions are redefined here:
https://github.com/makers-for-life/makair/blob/a6eced6016dda293cb0df3283ec5e83781fa0758/src/software/firmware/includes/parameters.h#L269-L270

I've verified in the log and as suspected got those warnings:
```
In file included from covid-respirator\src\software\firmware\includes\screen.h:19,
                 from Temp\arduino-sketch-0818B13F49F12784AC8B57DE7F1498F2\sketch\alarm_controller.cpp:17:
covid-respirator\src\software\firmware\includes\parameters.h:269: warning: "PIN_SERIAL_RX" redefined
  269 | #define PIN_SERIAL_RX PA12
      |
In file included from Arduino15\packages\STM32\hardware\stm32\1.8.0\system/STM32F4xx/stm32f4xx_hal_conf.h:4,
                 from Arduino15\packages\STM32\hardware\stm32\1.8.0\system/Drivers/STM32F4xx_HAL_Driver/Inc/stm32f4xx_hal.h:30,
                 from Arduino15\packages\STM32\hardware\stm32\1.8.0\system/Drivers/CMSIS/Device/ST/STM32F4xx/Include/stm32f4xx.h:250,
                 from Arduino15\packages\STM32\hardware\stm32\1.8.0\cores\arduino/stm32/stm32_def.h:34,
                 from Arduino15\packages\STM32\hardware\stm32\1.8.0\cores\arduino/stm32/clock.h:43,
                 from Arduino15\packages\STM32\hardware\stm32\1.8.0\cores\arduino/wiring_time.h:23,
                 from Arduino15\packages\STM32\hardware\stm32\1.8.0\cores\arduino/wiring.h:38,
                 from Arduino15\packages\STM32\hardware\stm32\1.8.0\cores\arduino/Arduino.h:36,
                 from covid-respirator\src\software\firmware\includes\parameters.h:12,
                 from covid-respirator\src\software\firmware\includes\screen.h:19,
                 from Temp\arduino-sketch-0818B13F49F12784AC8B57DE7F1498F2\sketch\alarm_controller.cpp:17:
Arduino15\packages\STM32\hardware\stm32\1.8.0\variants\NUCLEO_F4x1RE/variant.h:109: note: this is the location of the previous definition
  109 | #define PIN_SERIAL_RX           PA3
      |
In file included from covid-respirator\src\software\firmware\includes\screen.h:19,
                 from Temp\arduino-sketch-0818B13F49F12784AC8B57DE7F1498F2\sketch\alarm_controller.cpp:17:
covid-respirator\src\software\firmware\includes\parameters.h:270: warning: "PIN_SERIAL_TX" redefined
  270 | #define PIN_SERIAL_TX PA11
      |
In file included from Arduino15\packages\STM32\hardware\stm32\1.8.0\system/STM32F4xx/stm32f4xx_hal_conf.h:4,
                 from Arduino15\packages\STM32\hardware\stm32\1.8.0\system/Drivers/STM32F4xx_HAL_Driver/Inc/stm32f4xx_hal.h:30,
                 from Arduino15\packages\STM32\hardware\stm32\1.8.0\system/Drivers/CMSIS/Device/ST/STM32F4xx/Include/stm32f4xx.h:250,
                 from Arduino15\packages\STM32\hardware\stm32\1.8.0\cores\arduino/stm32/stm32_def.h:34,
                 from Arduino15\packages\STM32\hardware\stm32\1.8.0\cores\arduino/stm32/clock.h:43,
                 from Arduino15\packages\STM32\hardware\stm32\1.8.0\cores\arduino/wiring_time.h:23,
                 from Arduino15\packages\STM32\hardware\stm32\1.8.0\cores\arduino/wiring.h:38,
                 from Arduino15\packages\STM32\hardware\stm32\1.8.0\cores\arduino/Arduino.h:36,
                 from covid-respirator\src\software\firmware\includes\parameters.h:12,
                 from covid-respirator\src\software\firmware\includes\screen.h:19,
                 from Temp\arduino-sketch-0818B13F49F12784AC8B57DE7F1498F2\sketch\alarm_controller.cpp:17:
Arduino15\packages\STM32\hardware\stm32\1.8.0\variants\NUCLEO_F4x1RE/variant.h:110: note: this is the location of the previous definition
  110 | #define PIN_SERIAL_TX           PA2
      |
```

My understanding is the Hardware V2 will not use the Nucleo F4x1RE, Right?
And so you will use the Serial6 as the telemetry serial  and Serial as debug ?

As a first step, to avoid any confusion or problem I advise to rename those definition to ensure the right pins are used.
```suggestion
HardwareSerial Serial6(PIN_SERIAL6_RX, PIN_SERIAL6_TX);
```
and update in parameters.h the definition name.

As a second step, I propose to create a dedicated variant for your needs.
Advantages will be you could customize all your pin mappings and definitions and also manage easily hardware versions.
https://github.com/stm32duino/wiki/wiki/Custom-board-based-on-a-core

Reply:
This should be fixed in 06877206263f09da5dd3367f2e28e2e2b4621f18

> My understanding is the Hardware V2 will not use the Nucleo F4x1RE, Right?

We are using it but we have a PCB to connect the right pins to a micro USB port.

> And so you will use the Serial6 as the telemetry serial and Serial as debug ?

Yes!

Reply:
I'm a little bit lost with hardware version and the 'electronics' version.
Which one use a Nucleo connected to it or a integrated MCU... 😕

Reply:
In the codebase, `HARDWARE_VERSION` corresponds to the PCB version. Version 1 is currently used in produced machines. Version 2 exists but is not yet put in production. And there might be a version 3 in the future (that embeds the micro-controller), but this is not for sure.

So in this PR, I'm targeting version 2 (and above, hopefully) because version 1 does not have a telemetry-dedicated serial port.